### PR TITLE
[JENKINS-34228] Fixed IOException in DeleteBuildsCommand class

### PR DESCRIPTION
**Pull request by pjanouse**
Please review, thx. Pavel

### CODE REVIEW

File: core/src/main/java/hudson/cli/DeleteBuildsCommand.java [hunk 1]
```diff
@@ -54,10 +55,16 @@ protected void printUsageSummary(PrintStream stderr) {
     protected int act(List<AbstractBuild<?, ?>> builds) throws IOException {
         job.checkPermission(Run.DELETE);
 
-        for (AbstractBuild build : builds)
-            build.delete();
+        final HashSet<Integer> hs_builds = new HashSet<Integer>();
```
Comment (batmat) [inline core/src/main/java/hudson/cli/DeleteBuildsCommand.java]: NIT: java (local) variables with underscore is uncommon

File: core/src/main/java/hudson/cli/DeleteBuildsCommand.java [hunk 2]
```diff
@@ -54,10 +55,16 @@ protected void printUsageSummary(PrintStream stderr) {
...
 
-        stdout.println("Deleted "+builds.size()+" builds");
+        for (AbstractBuild build : builds) {
+            if (!hs_builds.contains(build.number)) {
+                build.delete();
+                hs_builds.add(build.number);
+            }
+        }
```
Comment (batmat) [inline core/src/main/java/hudson/cli/DeleteBuildsCommand.java]: Not sure I understand what you're trying to do here.
Reading the JIRA, I understand that an `IOException` can be thrown if we try to remove an already removed build. Shouldn't you just change the original code with a try/catch?

Or possibly improve/throw a subclass of IOException in https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/model/Run.java#L1480 to be better informed the to-be-removed build seems actually to be (already) absent?

**Comment by pjanouse** [comment]
Variable renamed.

Try/catch or change behavior of Run.delete() isn't good way IMO. 

I think, it isn't a bad way to ask to delete a build twice (either delete 5,5 or delete 1-5, 4-8). In both examples it is obvious and clear user asks to delete the build number 5. The issue is on the CLI delete-build impl. side - it asks the Jenkins model to delete such build twice which isn't correct - the first call is OK, the second should be skipped. That's the idea behind...

**Comment by pjanouse** [comment]
Fail is not related to this PR.

**Comment by batmat** [comment]
OK, just had a look at `AbstractBuildRangeCommand` and now I understand it better.

:+1: 

though a test would be great

**Comment by pjanouse** [comment]
Definitely, AbstractBuildRangeCommand class is covered by JENKINS-32874 and I'm finishing JENKINS-32777 where this case will be covered.

**Pull request merged**